samcli/lib/remote_invoke/sqs_invoke_executors.py

```diff
@@ -0,0 +1,146 @@
+    BotoActionExecutor,
+    RemoteInvokeIterableResponseType,
+    RemoteInvokeOutputFormat,
+    RemoteInvokeResponse,
+)
+
+LOG = logging.getLogger(__name__)
+QUEUE_URL = "QueueUrl"
+MESSAGE_BODY = "MessageBody"
+DELAY_SECONDS = "DelaySeconds"
+MESSAGE_ATTRIBUTES = "MessageAttributes"
+MESSAGE_SYSTEM_ATTRIBUTES = "MessageSystemAttributes"
+
+
+class SqsSendMessageExecutor(BotoActionExecutor):
+    """
+    Calls "send_message" method of "SQS" service with given input.
+    If a file location provided, the file handle will be passed as input object.
+    """
+
+    _sqs_client: SQSClient
+    _queue_url: str
+    _remote_output_format: RemoteInvokeOutputFormat
+    request_parameters: dict
+
+    def __init__(self, sqs_client: SQSClient, physical_id: str, remote_output_format: RemoteInvokeOutputFormat):
+        self._sqs_client = sqs_client
+        self._remote_output_format = remote_output_format
+        self._queue_url = physical_id
+        self.request_parameters = {}
+
+    def validate_action_parameters(self, parameters: dict) -> None:
+        """
+        Validates the input boto parameters and prepares the parameters for calling the API.
+
+        Parameters
+        ----------
+        parameters: dict
+            Boto parameters provided as input
+        """
+        try:
+            for parameter_key, parameter_value in parameters.items():
+                if parameter_key == QUEUE_URL:
+                    LOG.warning("QueueUrl is defined using the value provided for resource_id argument.")
+                elif parameter_key == MESSAGE_BODY:
+                    LOG.warning(
+                        "MessageBody is defined using the value provided for either --event or --event-file options."
+                    )
+                elif parameter_key == DELAY_SECONDS:
+                    self.request_parameters[parameter_key] = int(parameter_value)
+                elif parameter_key in {MESSAGE_ATTRIBUTES, MESSAGE_SYSTEM_ATTRIBUTES}:
+                    self.request_parameters[parameter_key] = json.loads(parameter_value)
+                else:
+                    self.request_parameters[parameter_key] = parameter_value
+        except (ValueError, JSONDecodeError) as err:
+            raise InvalidResourceBotoParameterException(f"Invalid value provided for parameter {parameter_key}", err)
+
+    def _execute_action(self, payload: str) -> RemoteInvokeIterableResponseType:
+        """
+        Calls "send_message" method to send a message to the SQS queue.
+
+        Parameters
+        ----------
+        payload: str
+            The MessageBody which will be sent to the SQS
+
+        Yields
+        ------
+        RemoteInvokeIterableResponseType
+            Response that is consumed by remote invoke consumers after execution
+        """
+        if payload:
+            self.request_parameters[MESSAGE_BODY] = payload
+        else:
+            self.request_parameters[MESSAGE_BODY] = "{}"
```

Comment:
A debug log could be useful to show that no payload is been provided therefore we are just giving a default empty one